src/components/checkbox/checkbox.js

```diff
@@ -68,11 +68,22 @@ function MdCheckboxDirective(inputDirective, $mdInkRipple, $mdAria, $mdConstant,
 
   function compile (tElement, tAttrs) {
 
+    var disabledParsed = $parse(tAttrs.ngDisabled) || angular.noop;
+
     tAttrs.type = 'checkbox';
     tAttrs.tabIndex = 0;
     tElement.attr('role', tAttrs.type);
 
     return function postLink(scope, element, attr, ngModelCtrl) {
+      scope.$watch(function () {
+        return typeof disabledParsed === 'function' && disabledParsed(scope);
```

Comment:
@robertmesserle - 
*  why are you check `typeof` here? L71 will always set a function, right?
*  the title to this PR does not match. Thoughts ?